src/stores/room-list/RoomListStore.ts

```diff
@@ -606,27 +664,41 @@ export class RoomListStoreClass extends AsyncStoreWithClient<IState> {
             // TODO: Remove debug: https://github.com/vector-im/element-web/issues/14602
             console.log("Adding filter condition:", filter);
         }
-        this.filterConditions.push(filter);
-        if (this.algorithm) {
-            this.algorithm.addFilterCondition(filter);
+        let promise = Promise.resolve(); // use a promise to maintain sync API contract
```

Comment:
Which contract of what are we maintaining as sync here...? The update trigger has become async by using a promise, so the comment feels unclear to me...

Reply:
This would be referencing the contract of the add/remove filter function itself. This should probably be bundled into the jsdoc instead of a random line comment - will move/clarify.